data/model/repository.py

```diff
@@ -701,6 +701,37 @@ def get_repository_size(repo_id: int):
         return 0
 
 
+def get_repository_sizes(repo_ids: list):
+    """
+    Returns a map from repository ID to the size of the repository in bytes.
+    List of repo_ids should be kept below 1000 to avoid performance issues.
+    """
+    if not repo_ids:
+        return {}
+
+    if len(repo_ids) > 1000:
+        logger.warning(
+            "Fetching more than 1000 repository sizes at once, you may experience performance issues."
+        )
```

Comment:
What is this function used for, wouldn't we want a way to paginate this instead?
Are the repo_ids restricted to a single namespace or is this a full table lookup?

Reply:
I could add pagination but for this particular use case it uses what ever the page size is for listing repositories. The repo_ids will belong to a single namespace the majority of the time but it's possible that they belong to separate namespaces. 

I could also replace this with individual requests for fetching each repository size but it looked like this gave better performance.